src/ORM/DataList.php

```diff
@@ -1310,35 +1376,43 @@ private function fetchEagerLoadManyMany(
      * Eager loading alleviates the N + 1 problem by querying the nested relationship tables before they are
      * needed using a single large `WHERE ID in ($ids)` SQL query instead of many `WHERE RelationID = $id` queries.
      *
-     * You can speicify nested relations by using dot notation, and you can also pass in multiple relations.
-     * When speicifying nested relations there is a maximum of 3 levels of relations allowed i.e. 2 dots
+     * You can specify nested relations by using dot notation, and you can also pass in multiple relations.
+     * When specifying nested relations there is a maximum of 3 levels of relations allowed i.e. 2 dots
      *
      * Example:
      * $myDataList->eagerLoad('MyRelation.NestedRelation.EvenMoreNestedRelation', 'DifferentRelation')
      *
-     * IMPORTANT: Calling eagerLoad() will cause any relations on DataObjects to be returned as an ArrayList
-     * instead of a subclass of DataList such as HasManyList i.e. MyDataObject->MyHasManyRelation() returns an ArrayList
+     * IMPORTANT: Calling eagerLoad() will cause any relations on DataObjects to be returned as an EagerLoadedDataList
+     * instead of a subclass of DataList such as HasManyList i.e. MyDataObject->MyHasManyRelation() returns an EagerLoadedDataList
      */
-    public function eagerLoad(...$relations): static
+    public function eagerLoad(...$relationChains): static
```

Comment:
This method now stores a list of all partial and complete eagerload relation chains into `eagerLoadAllRelations` to avoid duplicates, and a list of all _complete_ eagerload relation chains into `eagerLoadRelationChains` which `fetchEagerLoadRelations()` iterates over.